docs/examples/extract_slides.md

```diff
@@ -0,0 +1,113 @@
+# Data extraction from slides
+
+In this guide, we demonstrate how to extract data from slides.
+
+!!! tips "Motivation"
+
+   When we want to translate key information from slides into structured data, simply isolating the text and running extraction might not be enough. Sometimes the important data is in the images on the slides, so we should consider including them in our extraction pipeline.
+    
+## Defining the necessary Data Structures
+
+Let's say we want to extract the competitors from various presentations and categorize them according to their respective industries.
+
+Our data model will have `Industry` which will be a list of `Competitor`'s for a specific industry, and `Competition` which will aggregate the competitors for all the industries.
+
+```python
+from openai import OpenAI
+from pydantic import BaseModel, Field
+from typing import Optional, List
+
+class Competitor(BaseModel):
+    name: str
+    features: Optional[List[str]]
+
+
+# Define models
+class Industry(BaseModel):
+    """
+    Represents competitors from a specific industry extracted from an image using AI.
+    """
+
+    name: str = Field(
+        description="The name of the industry"
+    )
+    competitor_list: List[Competitor] = Field(
+        description="A list of competitors for this industry"
+    )
+
+class Competition(BaseModel):
+    """
+    This class serves as a structured representation of 
+    competitors and their qualities.
+    """
+
+    industry_list: List[IndustryCompetition] = Field(
+        description="A list of industries and their competitors"
+    )
+```
+
+## Competitors extraction
+
+To extract competitors from slides we will define a function which will read images from urls and extract the relevant information from them.
+
+```python
+import instructor
+from openai import OpenAI
+
+# Apply the patch to the OpenAI client
+# enables response_model keyword
+client = instructor.patch(
+    OpenAI(mode=instructor.Mode.MD_JSON)
```

Comment:
```suggestion
    OpenAI(), mode=instructor.Mode.MD_JSON
```

does this code run? lol

Reply:
Also why is it slides and not competitors? This is a different example.

Reply:
honestly it seemed that 'slides' was a better name because its more generic, i.e. this example is about extracting anything from a slide, not specifically competitors

I changed the name of the .py file to slides to make it symmetric, let me know if you agree

Reply:
the example above fails without specifying json mode
```

BadRequestError: Error code: 400 - {'error': {'message': '2 validation errors 
for Request\nbody -> function_call\n  extra fields not permitted 
(type=value_error.extra)\nbody -> functions\n  extra fields not permitted 
(type=value_error.extra)', 'type': 'invalid_request_error', 'param': None, 
'code': None}}

```